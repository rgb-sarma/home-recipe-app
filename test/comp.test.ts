import { beforeEach, expect, test } from "vitest";
import { setActivePinia, createPinia } from "pinia";
import { mount } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";
import HeaderBar from "../src/components/TEST/HeaderBar.vue";

test("mount component", async () => {
	beforeEach(() => {
		setActivePinia(createPinia());
	});

	expect(HeaderBar).toBeTruthy();

	const wrapper = mount(HeaderBar, {
		global: {
			plugins: [createTestingPinia()],
		},
		props: {
			count: 4,
		},
	});

	expect(wrapper.find(".test-count").text()).toContain("3 x 2 = 6");
	expect(wrapper.html()).toMatchSnapshot();

	await wrapper.find(".test-btn").trigger("click");

	expect(wrapper.find(".test-count").text()).toContain("3 x 3 = 9");

	await wrapper.find(".test-btn").trigger("click");

	expect(wrapper.find(".test-count").text()).toContain("3 x 4 = 12");
});
