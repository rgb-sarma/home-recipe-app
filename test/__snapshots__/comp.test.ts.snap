// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<div data-v-17198171=\\"\\" id=\\"fake-api\\" count=\\"4\\">
  <div data-v-17198171=\\"\\" class=\\"store-example\\">
    <div data-v-17198171=\\"\\" class=\\"counter\\">Counter: 0</div><button data-v-17198171=\\"\\">Increment store</button>
  </div>
  <div data-v-17198171=\\"\\" class=\\"test-count\\">3 x 2 = 6</div>
  <div data-v-17198171=\\"\\" class=\\"test-btn\\">x1</div>
</div>"
`;
